Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs

```diff
@@ -93,44 +96,84 @@ public override void OnDeactivate()
             }
 
+            InputSystem.onActionsChange -= BuildUI;
+
+            m_IsActivated = false;
+
             // Always end a session when deactivated.
             m_ActionEditorAnalytics?.End();
+
+            m_View?.DestroyView();
         }
 
         private void OnFocusIn(FocusInEvent @event = null)
         {
             if (!m_HasEditFocus)
             {
                 m_HasEditFocus = true;
-                OnEditFocus();
+                m_ActionEditorAnalytics.RegisterEditorFocusIn();
+                m_ActiveSettingsProvider = this;
+                SetIMGUIDropdownVisible(false, false);
             }
         }
 
-        private void OnFocusOut(FocusOutEvent @event = null)
+        void SaveAssetOnFocusLost()
         {
-            // This can be used to detect focus lost events of container elements, but will not detect window focus.
-            // Note that `event.relatedTarget` contains the element that gains focus, which is null if we select
-            // elements outside of project settings Editor Window. Also note that @event is null when we call this
-            // from OnDeactivate().
-            var element = (VisualElement)@event?.relatedTarget;
-            if (element != null || !m_HasEditFocus)
+#if UNITY_INPUT_SYSTEM_INPUT_ACTIONS_EDITOR_AUTO_SAVE_ON_FOCUS_LOST
+            var asset = GetAsset();
+            if (asset != null)
+                ValidateAndSaveAsset(asset);
+#endif
+        }
+
+        public static void SetIMGUIDropdownVisible(bool visible, bool optionWasSelected)
+        {
+            if (m_ActiveSettingsProvider == null)
                 return;
-            m_HasEditFocus = false;
-            OnEditFocusLost();
+
+            // If we selected an item from the dropdown, we *should* still be focused on this settings window - but
+            // since the IMGUI dropdown is technically a separate window, we have to refocus manually.
+            //
+            // If we didn't select a dropdown option, there's not a simple way to know where the focus has gone,
+            // so assume we lost focus and save if appropriate. ISXB-801
+            if (!visible && m_IMGUIDropdownVisible)
+            {
+                if (optionWasSelected)
+                    m_ActiveSettingsProvider.m_RootVisualElement.Focus();
+                else
+                    m_ActiveSettingsProvider.SaveAssetOnFocusLost();
+            }
+            else if (visible && !m_IMGUIDropdownVisible)
+            {
+                m_ActiveSettingsProvider.m_HasEditFocus = false;
+            }
+
+            m_IMGUIDropdownVisible = visible;
         }
 
-        private void OnEditFocus()
+        private async void DelayFocusLost(bool relatedTargetWasNull)
         {
-            m_ActionEditorAnalytics.RegisterEditorFocusIn();
+            await Task.Delay(120);
+
+            // We delay this call to ensure that the IMGUI flag has a chance to change first.
```

Comment:
Not keen on this 120 delay - is there a different way to achieve this without the async call?

Reply:
This is not related to this PR; will investigate if there was some previous merge issue.

Reply:
I believe this diff no longer exists on this branch, marking resolved